### PR TITLE
[WIP] Upgrade

### PR DESCRIPTION
**Pull request by khalidabuhakmeh**
This work will allow us to easily support `Range` for NEST. It has methods that can determine if range values are "date ranges" or "numeric ranges". This can be used to construct the necessary queries against Elasticsearch.

This will be a major version number update as I've upgraded to NEST 6.3

#39

### CODE REVIEW

File: src/Filter.Nest/FilterLogic.cs [hunk 1]
```diff
@@ -39,6 +39,12 @@ internal static class FilterLogic
                     var queries = new List<Func<QueryContainerDescriptor<T>, QueryContainer>>();
                     var aliasAttribute = validValueProperty.GetCustomAttribute<MappingAliasAttribute>();
 
+                    if (typeof(Range.Generic.IRange<>).IsAssignableFrom(filterProperty.PropertyType))
+                    {
+
```
Comment (tgharold) [inline src/Filter.Nest/FilterLogic.cs]: TODO item?

File: tests/Range.Tests/Generic/Range`1Tests.cs [hunk 2]
```diff
@@ -1,31 +1,95 @@
...
+            } as object;
+            
+            var result = Range.AsDateRange(dateTimeRange);
+            
+            Assert.Equal(2000, result.MinValue.Value.Year);
+            Assert.Equal(2020, result.MaxValue.Value.Year);
+        }
+        
+        [Fact]
+        public void Should_convert_object_that_is_datetimeoffset_range_to_datetimeoffset_range()
```
Comment (tgharold) [inline tests/Range.Tests/Generic/Range`1Tests.cs]: `Should_convert_object_that_is_datetime_range_to_datetimeoffset_range`

**Comment by billbogaiv** [comment]
I'm looking at the build failure on AppVeyor 👀.

**Comment by billbogaiv** [comment]
Since we're already upgrading a lot in this PR, I'm going to update the build script to use Cake. At least on my machine, the script-driven `MSBuild`-version won't mesh with the change to `4.6.2`...

```
CSC : error CS1617: Invalid option 'latest' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6. [C:\Users\bill.boga\Documents\projects\filter\src\Range\Range.csproj]
```

**Comment by billbogaiv** [comment]
Closing this in favor of: https://github.com/ritterim/filter/pull/56. The branch-name of the new PR should also work in conjunction with the switch-over to using GitVersion.

**Pull request closed without merging**